patches/ppc64le-and-riscv64-support.patch

```diff
@@ -97,10 +97,29 @@ index 6f8144b0954..8a8195e9b66 100644
  
  BUILD_TARGETS.forEach(buildTarget => {
 diff --git a/build/gulpfile.vscode.js b/build/gulpfile.vscode.js
-index e1507e0424f..80ed8d91a41 100644
+index 3b1aeafd080..c3f93d7a607 100644
 --- a/build/gulpfile.vscode.js
 +++ b/build/gulpfile.vscode.js
-@@ -429,6 +429,8 @@ const BUILD_TARGETS = [
+@@ -339,11 +339,17 @@ function packageTask(platform, arch, sourceFolderName, destinationFolderName, op
+ 			all = es.merge(all, shortcut);
+ 		}
+ 
++		const electronOverride = {};
++		if (platform === 'linux' && arch === 'riscv64') {
++			// electron doesn't support riscv64, so a fork is used here.
++			electronOverride.repo = 'riscv-forks/electron-riscv-releases';
```

Comment:
This can be achieved with `ELECTRON_MIRROR`, can it not?

Reply:
Yes.

Reply:
Not sure. `ELECTRON_MIRROR` seems to be a `electron-builder` variable.
VSCode seems to use `electronRepository` in the `product.json`

https://github.com/microsoft/vscode/blob/0fd9a006f7abcb48ef1e5a9f9ceeca33e1d1daaa/build/lib/electron.ts#L201

Reply:
> This can be achieved with `ELECTRON_MIRROR`, can it not?

@lex-ibm I didn't see `ELECTRON_MIRROR` anywhere in your PR? Where did you implemented it in your PR?

Reply:
> This can be achieved with `ELECTRON_MIRROR`, can it not?

@lex-ibm I didn't see `ELECTRON_MIRROR` anywhere in your PR? Where did you implemented it in your PR?

Reply:
Oh the network glitch caused me to send the comment twice. Sorry.

Reply:
Resolved by using a `VSCODE_ELECTRON_REPO` environment variable for the electron repo.